community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -163,12 +167,13 @@ TransactionType upgradeToSchemaTransaction() throws InvalidTransactionTypeKernel
     private final Collection<Command> extractedCommands = new ArrayCollection<>( 32 );
     private final Locks locksManager;
     private final boolean txTerminationAwareLocks;
+    private final boolean deferringLocks;
```

Comment:
We could do it the same way we do with `locksManager`.
`this.statementLocks = statementLocksManager.newClient()`
and hide `locksManager` in `statementLocksManager`